src/instructlab/rag/document_store.py

```diff
@@ -0,0 +1,53 @@
+"""
+A module to define interfaces to manage document stores:
+- Ingest documents in it. This typically involves creating the document embeddings through a configured embedding model.
+- Query the stored documents for an augmented RAG context. Advanced retrieval techniques can be used to improve the reliabiliyty of
+the generated context.
+"""
+
+# Standard
+from abc import ABC, abstractmethod
+
+
+class DocumentStoreRetriever(ABC):
+    """
+    Interface for retrieving documents from a document store.
+
+    Documents are stored as vector embeddings generated by a configured embedding model.
+    Retrieval is based on user queries, which are transformed into vector embeddings using the same embedding model
+    to fetch the matching documents.
+    """
+
+    @abstractmethod
+    def augmented_context(self, user_query: str) -> str:
+        """
+        Retrieve documents from the actual store matching the given `user_query` and compute the augmented context to be used
+        in a RAG chat pipeline.
+
+        Params:
+          user_query: The original user query.
+        Returns:
+          str: The augmented context to use in a RAG chat.
+        """
```

Comment:
Should we raise `NotImplementedError` or add `pass` since this is an interface?

Common approach: https://github.com/google/etils/blob/6ff7408dbec072e4561e393d5abac9625476df36/etils/epath/backend.py#L41

```suggestion
    @abstractmethod
    def augmented_context(self, user_query: str) -> str:
        """
        Retrieve documents from the actual store matching the given `user_query` and compute the augmented context to be used
        in a RAG chat pipeline.

        Params:
          user_query: The original user query.
        Returns:
          str: The augmented context to use in a RAG chat.
        """
        raise NotImplementedError
```

Reply:
I agree with you, but I based this implementation from the few interfaces I found, which do the same (see https://github.com/instructlab/instructlab/blob/main/src/instructlab/llamacpp/llamacpp_convert_to_gguf.py)
I'm available to change it, of course: should we involve someone to decide b/w `pass` or `NotImplementedError`?

Reply:
I think if this is not a norm in ilab, we should avoid this for now and keep the current implementation.